e2e_tests/tests/test_system.py

```diff
@@ -631,22 +631,32 @@ def _fetch_slots() -> List[Dict[str, Any]]:
     command = ["det", "-m", conf.make_master_url(), "agent", "disable", "--drain", agent_id]
     subprocess.check_call(command)
 
-    exp.wait_for_experiment_state(experiment_id, "COMPLETED")
-
     # Try to launch another experiment.
-    experiment_id = exp.create_experiment(
+    experiment_id_no_start = exp.create_experiment(
         conf.fixtures_path("no_op/single-medium-train-step.yaml"),
         conf.fixtures_path("no_op"),
         None,
     )
-    exp.wait_for_experiment_state(experiment_id, "ACTIVE")
+    exp.wait_for_experiment_state(experiment_id_no_start, "ACTIVE")
```

Comment:
i think to see the bug I'm concerned about, we need to quickly re-enable the slot after setting it to drain. then i think 2 experiments will land on the same slot momentarily

Reply:
how about now?

Reply:
what happens without the call to disable --drain the agent again?

Reply:
the 2nd experiment will start when the first one finishes.

Reply:
can you add that into the test? making sure the 2nd doesn't start while the 1st is active? sorry for asking for so much. that's just what i was concerned about from the start. we're not quite testing it yet.

Reply:
found a bug, x-post here
1. start a 2 slot cluster
2. start a 1 slot experiment with really high resources.weight so you dont it wont get displaced
3. disable --drain its agent
4. start a new plain-old low weight 1 slot experiment
5. it won't get scheduled because fair share thinks the cluster is at capacity (since it only knows of 1 agent with 1 slot and has already offered a slot to the first one)